### ISSUE TITLE
keygen2 for rsa, dsa, ecdsa

### ISSUE THREAD

**Issue by nikolaynazarko** [ewpa/LibSSH-ESP32#3]
Hi. Thanks a lot for your efforts in this project. 
I just want to ask is keygen2 support rsa, dsa, ecdsa key types? 
Because when I try RSA I got  id_rsa file with 0 size(I've added some code for browse files ):

![keys](https://user-images.githubusercontent.com/54119989/87434478-38451e00-c5f3-11ea-8fd8-bf35ce007021.png)

Same results for dsa, ecdsa. Only ed25519 working on my side.
Best regards.

**Comment by ewpa** [comment]
Hi, I looked at this for some time in the past while trying to create a host key directly.  The combination of libssh and mbedtls do not have the implementations needed to export the keys other than for the ed25519 type.  To confirm, I built this port on Linux using the same build configuration and get the same error.
```
$ ./examples/keygen2 --type rsa --bits 1024 --file ./id_rsa
Error: Failed to write private key file
```
So this is an existing limitation and there is no fix in scope.
